src/mem/alloc.h

```diff
@@ -794,16 +805,16 @@ namespace snmalloc
 
     template<Boundary location>
     static uintptr_t
-    external_pointer(void* p, uint8_t sizeclass, size_t end_point)
+    external_pointer(void* p, uint8_t sizeclass, void* end_point)
     {
       size_t rsize = sizeclass_to_size(sizeclass);
-      size_t end_point_correction = location == End ?
-        (end_point - 1) :
-        (location == OnePastEnd ? end_point : (end_point - rsize));
-      size_t offset_from_end =
-        (end_point - 1) - static_cast<size_t>(address_cast(p));
-      size_t end_to_end = round_by_sizeclass(rsize, offset_from_end);
-      return end_point_correction - end_to_end;
+      void *end_point_correction = location == End ?
+        ((char*)end_point - 1) :
```

Comment:
Can we use `uint8_t*` rather than `char*`.  I find this more descriptive.

Reply:
Since we're doing addressable-unit calculations, I think `char*` is technically more correct here, though I doubt anyone will try to use snmalloc on 16-bit addressable platforms (mind you, I might have said the same thing about platfroms with 128-bit fat pointers...).

Reply:
If we aren't a byte addressable platform, I have no idea what would happen with this code base.  

`uint8_t` is in some sense "more" correct, as it codifies an assumption. :-)